pandas/_libs/groupby_helper.pxi.in

```diff
@@ -360,11 +361,19 @@ def group_last_{{name}}(ndarray[{{dest_type2}}, ndim=2] out,
         raise AssertionError("len(index) != len(labels)")
 
     nobs = np.zeros((<object> out).shape, dtype=np.int64)
+    {{if name=='object'}}
+    resx = np.empty((<object> out).shape, dtype=object)
```

Comment:
Was really hoping to not even have this conditional, but when trying `resx = np.empty_like(out)` and even `resx = np.empty_like(out, dtype='object')` it kept SegFaulting on objects